src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -1071,17 +1071,20 @@ trait Implicits {
      *  bound, the implicits infos which are members of these companion objects.
      */
     private def companionImplicitMap(tp: Type): InfoMap = {
+      val isScala213 = settings.isScala213
```

Comment:
I wonder if we can sanely cache this (and its friend) per-run. It's not a _lot_ of work to compare `settings.source.value` with a cached 2.13 version, but it's used a lot.

Reply:
I think that we should be more daring - 
Even with some of (sometimes unsafe and inaccurate) current caching that the are several Million checks to Boolean settings in the compile of a 5 second akka actor corpus. I did have a look at using method handles which provide a more aggressive git based optimisation on the basis that you don't often change most of these setting on a per-run basis. 
I had a few ideas on fixing it but have been focussed on other gains recently. 
Happy to share ideas if someone want to progress this, but these optimisations are very sensitive to the java version that you are running (Java 8, 9 and 10 all have different tradeoffs - then I started looking at easier gains elsewhere)